cpp/src/strings/regex/regex.cuh

```diff
@@ -143,21 +124,83 @@ class reprog_device {
    */
   [[nodiscard]] __device__ inline bool is_empty() const;
 
+  /**
+   * @brief Returns the size needed for working memory for the given thread count.
+   *
+   * @param num_threads Number of threads to be executed in parallel
+   * @return Size of working memory in bytes
+   */
+  [[nodiscard]] std::size_t working_memory_size(int32_t num_threads) const;
+
+  /**
+   * @brief Compute working memory for the given thread count with a maximum size.
+   *
+   * The `min_rows` overrules the `requested_max_size`.
+   * That is, the `requested_max_size` may be
+   * exceeded to keep the number of rows greater than `min_rows`.
+   * Also, if `rows < min_rows` then `min_rows` is not enforced.
+   *
+   * @param rows Number of rows to execute in parallel
+   * @param min_rows The least number of rows to meet `max_size`
+   * @param requested_max_size Requested maximum bytes for the working memory
+   * @return The size of the working memory and the number of parallel rows it will support
+   */
+  [[nodiscard]] std::pair<std::size_t, int32_t> compute_strided_working_memory(
+    int32_t rows,
+    int32_t min_rows               = MINIMUM_THREADS,
+    std::size_t requested_max_size = MAX_WORKING_MEM) const;
```

Comment:
```suggestion
    int32_t const rows,
    int32_t const min_rows               = MINIMUM_THREADS,
    std::size_t const requested_max_size = MAX_WORKING_MEM) const;
```

Reply:
Probably can make a few other function parameters in this file constant, but I guess they're mostly host functions so it's not a huge deal.

Reply:
I don't think declaring primitive parameter types that are passed by value `const` makes much difference and does not communicate anything special to the caller.

Reply:
It can make a difference in device code where inlined functions can reuse the same values in registers from previous functions if they're all const, which in turn reduces memory traffic and register pressure, but I realized after I wrote this suggestion that this is host code and not performance-critical which is why I discounted it in my second comment. We can resolve this.